### DISCUSSION TITLE
Action to jump to the next black screen

### DISCUSSION THREAD

**Discussion by jrporter** [mifi/lossless-cut#2077] [Ideas]
I am using Losslesscut to take a large number of MKV videos which are not broken up by chapters and manually insert chapter markers at locations which correspond to black screens. My current workflow is to manually scrub to find the desired location, and then press "b" to split the segment at that location, which works but is tedious. The current feature for detecting black screens is not ideal for my purposes for a couple reasons: 1) I don't want instances of black cut from my exported video, which happens after I "invert" the segments without manual steps after, and b) often times I get clusters of undesired segments during credits sections from the black between consecutive text cards.

What would be really useful is if there was an action that would simply jump the scrubber to the midpoint of the next (relative to the current location) detected portion of black screen. That way, for users working through my use case, the flow would be to simply trigger that action, and either cut at that location if desired, or trigger the action to keep moving if it was a false positive.

To reduce the number of speed bumps, this would probably also mean that the configurable parameters for black screen detection would need to be persistent and go into the app settings, rather than being a dialog that appears every time the action is triggered.

Is this the kind of thing that could be considered for a future release?

**Comment by mifi** [comment]
> 1) I don't want instances of black cut from my exported video, which happens after I "invert" the segments without manual steps after

if you select Export mode: Segments to chapters, nothing will be cut away in your exported video. segments will simply be converted into chapters. isn't this what you want?

b) often times I get clusters of undesired segments during credits sections from the black between consecutive text cards.

I believe you can etiher
- delete these clusters of segments that you don't need
- exclude the credits from the black detect, by creating a single segment that spans the whole range you want to detect black scenes (but ends before the credits), then running detect black scenes (it will only detect for the current segment)

doesn't that solve the problem?

**Reply by jrporter** [reply]
> if you select Export mode: Segments to chapters, nothing will be cut away in your exported video. segments will simply be converted into chapters. isn't this what you want?

Yeah, my apologies, I misstated that part about black being cut. But "segments to chapters" still doesn't give me the desired behavior. The issue is that when I run black detect, it outputs sections of nonzero length. Then, I invert so that the segments are the actual video content, with non-segment 'gaps' between them. When I export, with segments to chapters, it still treats the portions between the segments as chapters.

So as a hypothetical, let's say I have a 22 minute video file. It's made up of 2 sections of continuous video roughly equal in length, and there is a 'fade to black' break between them. Doing the above process yields an exported file with 3 chapters (segment 1, the black segment, and segment 2) when the desired behavior is just 2 chapters. When a video has multiple breaks it, this effect is amplified, with more of these 'ghost chapters' so to speak being created. When I am playing the output file, I want every chapter marker to be useful and jump to a block of content, but N-1 of them instead jump to a black section. It's kind of awkward redundancy.

The best possible way, from a user experience perspective, to support this use case would be to give the user to options in the detect black settings. 1) create segments bounding the detected black with everything else treated as a non-segment (current behavior), or 2) cut the current segment into subsegments with the cutpoints being centered in each black section. Behavior #1 is fantastic if your primary use case is to split up a video file into multiple smaller chunks. But behavior #2 is what's needed if you are trying to create chapters for segments in a television show, for example, which map to what was shown between commercial breaks.

But I wasn't sure how feasible it would be to introduce another behavior option in the settings, so I thought it might be more straightforward to implement a triggered action that would just jump to the next detected black. Not an automated solution, but would still be a nice accelerator for users who are ultimately trying to do this task.

> doesn't that solve the problem?

"Solve" is relative. I can do that, but it's tedious, so not the most friendly workaround when you are trying to go through hundreds and hundreds of videos.

**Reply by mifi** [reply]
I think it's easier to just add two modes to the black / silence detech tools like you explained. I will implement that because I agree it makes a lot of sense to create segments *between* the black/silent areas, similar to how the scene change filter works (it creates segments between scene changes)
